test/e2e/config_test.go

```diff
@@ -92,6 +94,50 @@ func TestClusterMonitoringOperatorConfiguration(t *testing.T) {
 	t.Log("asserting that CMO goes back healthy after the configuration is fixed")
 	assertOperatorCondition(t, configv1.OperatorDegraded, configv1.ConditionFalse)
 	assertOperatorCondition(t, configv1.OperatorAvailable, configv1.ConditionTrue)
+
+	// Push a configuration that disables Grafana.
+	validCM.Data["config.yaml"] += `grafana: {"enabled": false}`
+
+	if err := f.OperatorClient.CreateOrUpdateConfigMap(validCM); err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for Grafana deployment to disappear.
+	err = framework.Poll(time.Second, 5*time.Minute, func() error {
+		_, err := f.KubeClient.AppsV1().Deployments(f.Ns).Get(context.TODO(), "grafana", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+
+		return errors.New("Grafana deployment still exists")
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log("asserting that CMO is healthy after disabling Grafana")
+	assertOperatorCondition(t, configv1.OperatorDegraded, configv1.ConditionFalse)
+	assertOperatorCondition(t, configv1.OperatorAvailable, configv1.ConditionTrue)
+
+	// Push a default configuration that re-enables Grafana.
+	validCM.Data["config.yaml"] = "enableUserWorkload: true"
```

Comment:
How does enabling user workload monitoring also enable grafana?

Reply:
It doesn't. That was just setting the config back to what it was originally, i.e. removing the previously added Grafana bit. It was definitely confusing though. The new test is different.